agent/util-scripts/pbench-tool-meister-start

```diff
@@ -731,13 +733,86 @@ def main(_prog: str, cli_params: Namespace) -> int:
         ssh_opts=ssh_opts,
     )
 
+    redis_server_spec = cli_params.redis_server
+    tds_server_spec = cli_params.tool_data_sink
+
     if cli_params.orchestrate == "create":
         orchestrate = True
         ssh_cmd = "ssh"
         ssh_path = find_executable(ssh_cmd)
         if ssh_path is None:
             logger.error("required ssh command not in our PATH")
             return ReturnCode.MISSINGSSHCMD
+
+        # Connect with each of the tool hosts: if we can't connect via ssh,
+        # this isn't going to work, so if any of them fails we'll terminate
+        # with an error code.
+        #
+        # If we can connect, they'll tell us the IP address from which they see
+        # us connecting, and we'll use that (by default) as the server address.
+        localhost = LocalRemoteHost()
+        origin_ip = set()
+        template = TemplateSsh(ssh_cmd, shlex.split(ssh_opts), ['echo', '${SSH_CONNECTION}'])
+        for host in tool_group.hostnames.keys():
+            if not localhost.is_local(host):
+                template.start(host)
+
+        for host, params in tool_group.hostnames.items():
+            if not localhost.is_local(host):
+                connection = template.wait(host)
+                logger.debug("Host %s reports connection `%s`", host, connection)
+                if connection[0] == 0 and connection[1]:
+                    # The SSH_CONNECTION value is the full `stdout` and we
+                    # don't expect a stderr on success; the format is
+                    #   "origin_node origin_port local_node local_port"
+                    # we only need the origin node because we know that's an
+                    # IP for our local TDS and Redis host that the remote can
+                    # understand.
+                    origin = connection[1].split()[0]
+                    origin_ip.add(origin)
+
+                    # Store the origin reported by this remote in its parameter
+                    # block: we'll program this as the connection address when
+                    # we send the init handshake to that remote later.
+                    params["origin_host"] = origin
+                else:
+                    # If `ssh` fails, there's no point in continuing!
+                    logger.error("Host %s reports %s", host, connection)
```

Comment:
This is going to log a stringized tuple...is that really what you want?  (Wouldn't it be clearer to unpack and annotate it a bit?)